### PR TITLE
Migration to TS2.0

### LINKED ISSUE #67: TypeScript 2.0 for dojo/loader

### PR DESCRIPTION
**Pull request by agubler**
* [x] Upgrade to TS2.0 (beta)
 * [x] Add `strictNullChecks`
 * [x] Add `noImplicitThis `

Fixes #67

### CODE REVIEW

File: src/loader.ts [hunk 1]
```diff
@@ -216,10 +216,12 @@
 					packageDescriptor.location = packageDescriptor.location.replace(/\/*$/, '/');
 				}
 
-				config.pkgs[packageDescriptor.name] = packageDescriptor;
+				if (config && config.pkgs) {
```
Comment (kitsonk) [inline src/loader.ts]: Does this avoid the typecast?

```typescript
if (config && config.pkgs && packageDescriptor.name) {
    config.pkgs[packageDescriptor.name] = packageDescriptor;
}
```

Comment (agubler) [inline src/loader.ts]: Ah yes it probably does, thanks.

**Comment by kitsonk** [comment]
👍

File: src/loader.ts [hunk 2]
```diff
@@ -330,16 +334,16 @@
 	let loadNodeModule: (moduleId: string, parent?: DojoLoader.Module) => any = noop;
 
 	function contextRequire(moduleId: string, unused?: void, referenceModule?: DojoLoader.Module): DojoLoader.Module;
-	function contextRequire(dependencies: string[], callback: DojoLoader.RequireCallback, referenceModule?: DojoLoader.Module): DojoLoader.Module;
-	function contextRequire(dependencies: string | string[], callback: any, referenceModule?: DojoLoader.Module): DojoLoader.Module {
-		let module: DojoLoader.Module;
+	function contextRequire(dependencies: string[], callback: DojoLoader.RequireCallback | undefined, referenceModule?: DojoLoader.Module): DojoLoader.Module;
+	function contextRequire(dependencies: string | string[], callback: any, referenceModule?: DojoLoader.Module): DojoLoader.Module | undefined {
```
Comment (agubler) [inline src/loader.ts]: this can just be an optional argument

File: src/loader.ts [hunk 3]
```diff
@@ -393,7 +397,7 @@
 		return result;
 	}
 
-	function runMapProgram(targetModuleId: string, map: DojoLoader.MapItem[]): DojoLoader.MapSource {
+	function runMapProgram(targetModuleId: string, map: DojoLoader.MapItem[] | undefined): DojoLoader.MapSource | undefined {
```
Comment (agubler) [inline src/loader.ts]: could be an optional parameter

**Pull request merged**